playbooks/ceph.yml

```diff
@@ -357,6 +357,8 @@
       ansible.builtin.import_role:
         name: cifmw_cephadm
         tasks_from: pre
+      vars:
+        cifmw_cephadm_bootstrap_conf: ""
```

Comment:
`cifmw_cephadm_bootstrap_conf: /tmp/initial_ceph.conf ` is done in [1] , so the conf file generation from template at [2] is never executed.
moreover the code in [3] and [4]  relies on cifmw_cephadm_bootstrap_conf, so i had to unset it only for pre.yaml .
We shall discuss and do it in a better way.

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L253
[2] https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_cephadm/tasks/pre.yml#L147
[3]  https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L324
[4]  https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L396

Reply:
> `cifmw_cephadm_bootstrap_conf: /tmp/initial_ceph.conf ` is done in [1] , 
> so the conf file generation from template at [2] is never executed 
> and the `/home/ceph-admin/assimilate.conf` in the compute node will always have `initial_ceph.conf`

Yes, you're right. When ci-framework is used we always expect the initial_ceph.conf to be passed directly and created by another role, so we have no concept of supporting additional overrides via `ceph_conf_overrides`.
 
> moreover the code in [3] and [4] relies on cifmw_cephadm_bootstrap_conf, 

We don't have a problem that I know of in the code for [3] and [4]. It works and the items in the initial ceph.conf are looked up successfully in order to get the networks. It works because we're always passing ` cifmw_cephadm_bootstrap_conf`.

 > so i had to unset it only for pre.yaml . We shall discuss and do it in a better way.

What do you mean you had to unset it in pre.yml? Do you mean you had to (literally or effectively) remove this part of (lines 148-150) of TASK `Generate cifmw_cephadm_assimilate_conf on bootstrap node`?

```yaml
  when:
    - cifmw_cephadm_bootstrap_conf is not defined or
      cifmw_cephadm_bootstrap_conf | length == 0
```

The change I see in pre.yml is that it's switched from a `config_template` to a `ansible.builtin.template`.


> [1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L253
>  [2] https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_cephadm/tasks/pre.yml#L147
>  [3] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L324 
> [4] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L396

Reply:
Thanks for your thoughts.
yeah my only worry is about this task at [1] being skipped due to cifmw_cephadm_bootstrap_conf

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_cephadm/tasks/pre.yml#L149

but like i you mentioned your other comment, if we create another task rendering the overrides only for rgw in rgw.yaml, there is no confustion here.

Reply:
The problem is if we want support or not a `cephconfigoverride` interface. If we want to rely on a generic mechanism that apply overrides that are not necessarily part of rgw, then we should improve the existing code to handle them. 
If the plan is to not support it and **only** optimize the RGW config option generation, then we can move that code in rgw.yaml.
If in the future we need a generic mechanism to handle overrides, we need to rework this patch.

Reply:
Per our conversation we do NOT support the cephconfigoverride interface.

Reply:
Per our conversation the generic mechanism is via 

https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_ceph_spec/templates/initial_ceph.conf.j2